### PR TITLE
mongos is also valid target for stepping down and numactl

### LINKED ISSUE #137: Drop root privileges when running mongos
The entrypoint script drops root privileges when running `mongod`, but not when running `mongos`. This causes startup warnings in sharded clusters which are then printed on connecting clients:

```
I CONTROL  [main] ** WARNING: You are running this process as the root user, which is not recommended.
```

### PR DESCRIPTION
**Pull request by yosifkit**
Fixes #137

`numctl` is necessary for `mongos`.
> you should use numactl to start your mongod instances, including the config servers, mongos instances, and any clients.
> \- [docs](https://docs.mongodb.com/manual/administration/production-notes/#mongodb-and-numa-hardware)

Maybe we should also add `numctl` when they do `mongo` for the client or even `mongodump` and friends?

```console
$ # [[ "$1" == mongo* ]] would match these ones:
mongo         mongodump     mongofiles    mongooplog    mongorestore  mongostat
mongod        mongoexport   mongoimport   mongoperf     mongos        mongotop
```

Should we also `chown` and `gosu mongodb` on all of them as well?

### CODE REVIEW

**Comment by tianon** [comment]
Hmm, we should definitely `numactl` all those IMO, since they're all "clients":

- `mongo`: "the" client for MongoDB
- `mongodump`: client for dumping data
- `mongofiles`: client for interacting with GridFS
- `mongooplog`: client for connecting to remote server _and_ local server to sync
- `mongorestore`: the other half of `mongodump` (client)
- `mongostat`: "status" of server (client)
- `mongod`: the server itself
- `mongoexport`: client for exporting data
- `mongoimport`: opposite of `mongoexport` (client)
- `mongoperf`: this is basically `bonnie++`, but specific to MongoDB! (_not_ a client!)
- `mongos`: "sharding" frontend for the server
- `mongotop`: client for gathering stats of a server (client)

So in conclusion, everything in `mongo*` _is_ either a client or a server _except_ `mongoperf`, but it's technically a fake server that does nothing but heavy IO, so it should be safe inside `numactl` too (and good)

Regarding `chown` -- that's probably best left to just `mongod`, since as far as I can tell, `mongos` doesn't use either of those `/data` directories.

`gosu` should be fine and good, though.

**Comment by yosifkit** [comment]
@tianon, updated! :tada:

**Comment by tianon** [comment]
:+1:
